app/views/handlers/list_action.py

```diff
@@ -47,7 +47,13 @@ def get_section_summary_url(self):
 
     def get_next_location_url(self):
         if self._return_to == "section-summary":
-            return self.get_section_summary_url()
+            if self.router.can_display_summary_for_section(
```

Comment:
Not sure this is all necessary, I can possibly understand the completeness check in can_display_summary_for_section, but checking for a summary seems unnecessary as  you can't have a return_to = section_summary unless it came from the section summary. The only other way would be playing with URLs, which all bets are then off

Reply:
This is needed for a situation where all elements from a list are removed, if I remove all people in test_list_collector than summary with no householders is not displayed and I go back to list collector initial question..